lambdas/data-migration1/src/rules.ts

```diff
@@ -0,0 +1,162 @@
+import Knex from 'knex';
+
+import DynamoDbSearchQueue from '@cumulus/aws-client/DynamoDbSearchQueue';
+import { CollectionRecord, RuleRecord } from '@cumulus/db';
+import { envUtils } from '@cumulus/common';
+import Logger from '@cumulus/logger';
+
+import { RecordDoesNotExist } from '@cumulus/errors';
+import { RecordAlreadyMigrated } from './errors';
+import { MigrationSummary } from './types';
+
+const logger = new Logger({ sender: '@cumulus/data-migration/rules' });
+const Manager = require('@cumulus/api/models/base');
+const schemas = require('@cumulus/api/models/schemas');
+
+/**
+ *
+ * Retrieve cumulusId from Provider using id.
+ *
+ * @param {string} name - Provider ID
+ * @param {Knex} knex - Knex client for writing to RDS database
+ * @returns {Promise<number>} - Cumulus ID for record
+ * @throws {RecordDoesNotExist} if record cannot be found
+ */
+const getProviderCumulusId = async (
+  name: string,
+  knex: Knex
+): Promise<any> => {
+  const record = await knex.queryBuilder()
+    .select()
+    .table('providers')
+    .where({ name: name })
+    .first();
+
+  if (record === undefined) {
+    throw new RecordDoesNotExist(`Provider with id ${name} does not exist.`);
+  }
+  return record;
+};
+
+/**
+ *
+ * Retrieve cumulusId from Collection using name and version.
+ *
+ * @param {string} name - Collection Name
+ * @param {string} version - Collection version
+ * @param {Knex} knex - Knex client for writing to RDS database
+ * @returns {Promise<number>} - Cumulus ID for record
+ * @throws {RecordDoesNotExist} if record cannot be found
+ */
+const getCollectionCumulusId = async (
+  name : string,
+  version: string,
+  knex: Knex
+): Promise<number> => {
+  const record = await knex.select(knex.ref('cumulusId').as('collectionCumulusId'))
+    .from<CollectionRecord>('collections')
+    .where({ name: name, version: version })
+    .first();
+
+  if (record === undefined) {
+    throw new RecordDoesNotExist(`Collection with name ${name} and version ${version} does not exist.`);
+  }
+  return record.collectionCumulusId;
+};
+
```

Comment:
I was considering having a generic `getCumulusId` function instead of `getCollectionCumulusId` and `getProviderCumulusId`